corehq/apps/registration/utils.py

```diff
@@ -179,7 +180,8 @@ def request_new_domain(request, form, org, domain_type=None, new_user=True):
         dom_req.activation_guid = uuid.uuid1().hex
 
     new_domain = Domain(
-        name=form.cleaned_data['domain_name'],
+        name='',
```

Comment:
Ok but then I would say just leave the name parameter out all together or even better instead of having to save twice below you could use `Domain.get_db().server.next_uuid(count=1)` to get the value for the ID and set both `_id` and `name`.